capstone/src/main/java/com/google/sps/servlets/StudySetServlet.java

```diff
@@ -49,6 +54,16 @@
     "study_set.title ILIKE ? OR study_set.description ILIKE ? OR user_info.user_name ILIKE ? " +
     "GROUP BY study_set.id, university.id, user_info.id";
 
+  private final String INSERT_CARD_STATEMENT =
+    "INSERT INTO CARD (study_set_id, front, back) VALUES(?, ?, ?);";
```

Comment:
Currently you're running 1 insert for the study set (this is fine), 1 select to get the new ID, and N inserts for each card. Each of these statements can take a while, so it's good to ask whether you can cut down the number of queries. I think the SELECT and all the card INSERTS could be combined into one query. Try adding a `RETURNING id` to the study set insert as shown at the bottom of this article: https://www.postgresqltutorial.com/postgresql-serial/. Then try inserting all the cards with a single query:

```
INSERT INTO CARD (study_set_id, front, back) VALUES(?, ?, ?), (?, ?, ?), (?, ?, ?), ...;
```